### DISCUSSION TITLE
[Request] Giving items when inventory is full. (Queue)

### DISCUSSION THREAD

**Discussion by mormoon** [Zrips/CMI#129] [Feature Requests]
When giving an item via /cmi give there is always a change that there inventory is "full" and the remaining items are "lost". We are running a shop that lets players buy certain items, sometime's the player did not get the item because there inventory was full at the time he bought the item. 

Since inventory's are being "managed" by CMI, could we introduce a new feature that will queue new items? Perhaps a variation of the cmi give
cmi give {name} name_tag 1 n &7Retour_Nether_1e_klas l &3{name}/n&d{date}_{time} **-q** to enable the queue.

The receiving player receives a message that {item} {number} is waiting to be delivered and that {player} need to free up {x} slots in order to receive that item. Optional a list is being shown if multiple items are queued, whereas the item can be removed from the queue. 

Staff should be able to view the queue for all players and items, either via a "logfile" or ingame. As an alternative, we could introduce a life timer that counts backwards. The timer should persist between server reloads. When the timer expires, the item should be removed from the queue. The player should be periodically be informed that items are waiting in the queue. 

Tnx for taking the time reading through this request.

**Comment by Zrips** [comment]
Would it be simpler just to drop item on ground? Well atleast for me it would be :) We can try to add feature like this in future.

**Comment by MCTyler** [comment]
That's using a lot of Server resources for Players to stupid to empty their inventory before buying things it seems. I agree with @Zrips Maybe a toggle for this so we can opt out of all this item tracking?

**Comment by mormoon** [comment]
Buying a item was just an example, a 2nd example would be, giving everyone 1 special event item of some sorts. It not reasonable to check everyone's inventory. If we can somehow track that the item is being dropped on the floor (print to console for evidence) and notify the player that would be a improvement. A small downside is that this only works if the player is online but fortunately we have the !ifonline special command.  Also me might need to "protect" the item in case someone else picks it up. (crowded places)

**Comment by mrfloris** [comment]
https://youtu.be/dhNFn7LP91A

In my opinion this ticket can get closed now. The CMI /give feature has been extended a while ago to drop the items on the floor when there is no free space in the inventory. It's the responsibility of the player to pay attention and pick up dropped items. And this is what is common on many servers. Unlike a queue.
